File: algorithms/bubble_sort/java/GenericBubbleSort.java

class GenericBubbleSort {

    public static <T extends Comparable<T>> void bubble_sort(T array[]){
        boolean swapped;
        int indexOfLastUnsortedElement =  array.length;
        do{
            swapped = false;
            for(int i=0; i < indexOfLastUnsortedElement - 1; i++){
                if(array[i].compareTo(array[i+1]) > 0){
                    swap(i, array);
                    swapped = true;
                }
            }
            indexOfLastUnsortedElement--;
            printArray(array);
        }while(swapped);
    }

    private static <T> void swap(int index, T[] array){
        T temp = array[index];
        array[index] = array[index + 1];
        array[index + 1] = temp;
    }

    private static <T> void printArray(T[] input){
        for(int i  = 0; i < input.length; i++){
            System.out.print(input[i] + " ");
        }
        System.out.print("\n");
    }

    public static void main(String[] args){
        bubble_sort(new Integer[] {4, 2, 9, 6, 23, 12, 34, 0, 1});
        bubble_sort(new String[] { "c", "a", "e", "d", "b" });
    }
}

File: algorithms/bubble_sort/csharp/GenericBubbleSort.cs
using System;

namespace BubbleSort
{
    class Program
    {
        static void Main(string[] args)
        {
            var random = new Random();
            int[] randomNumbers = new int[10];
            for (int i = 0; i < randomNumbers.Length; i++)
                randomNumbers[i] = random.Next(0, 20);

            Console.WriteLine("Current Array: {0}", string.Join(", ", randomNumbers));

            var sorted = randomNumbers.BubbleSort();
            Console.WriteLine("Bubble Sorted Array: {0}", string.Join(", ", sorted));
            Console.WriteLine("Hello World!");
        }
    }

    static class BubbleSortExtensions
    {
        public static T[] BubbleSort<T>(this T[] arr) where T : IComparable
        {
            int size = arr.Length;
            bool swapped = false;
            T temp;
            for (int i = 0; i < size - 1; i++)
            {
                swapped = false;
                for (int j = 0; j < size - 1 - i; j++)
                {
                    if (arr[j].CompareTo(arr[j + 1]) > 0)
                    {
                        temp = arr[j];
                        arr[j] = arr[j + 1];
                        arr[j + 1] = temp;
                        swapped = true;
                    }
                }

                if (!swapped) break;
            }
            return arr;
        }
    }
}
